agent/exec/container/adapter.go

```diff
@@ -61,15 +62,28 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
+
 	for {
 		if err := dec.Decode(&m); err != nil {
 			if err == io.EOF {
 				break
 			}
 			return err
 		}
-		// TODO(stevvooe): Report this status somewhere.
-		logrus.Debugln("pull progress", m)
+		if spamLimiter.Allow() {
+			if progress, ok := m["progressDetail"].(map[string]interface{}); ok {
+				// only post pull progress to log every couple of seconds
+				log.G(ctx).WithFields(logrus.Fields{
+					"id":      m["id"],
```

Comment:
Do these map lookups need to be checked? I suspect that having a `nil` value in the `Fields` map is okay, but let's make sure.